packages/frontend/core/src/modules/i18n/context.ts

```diff
@@ -0,0 +1,14 @@
+import { useService } from '@toeverything/infra';
+import { type PropsWithChildren, useEffect } from 'react';
+
+import { I18nService } from './services/i18n';
+
+export function I18nProvider({ children }: PropsWithChildren) {
+  const i18n = useService(I18nService).i18n;
+
+  useEffect(() => {
+    i18n.init();
```

Comment:
Should we wait for i18n to be ready before rendering children? there might be flickering from en to other languages?

Reply:
en better than nothing